src/examples/rpp/doxygen/retry_when.cpp

```diff
@@ -0,0 +1,63 @@
+#include <rpp/rpp.hpp>
+
+#include <iostream>
+#include <string>
+
+/**
+ * @example retry_when.cpp
+ **/
+
+int main()
+{
+    //! [retry_when delay]
+    size_t retry_count = 0;
+    rpp::source::create<std::string>([&retry_count](const auto& sub) {
+        if (retry_count != 4)
+        {
```

Comment:
coderabbit.ai thinks there is a missing increment here, but I can no longer see its comments, so I hereby re-add the comment by hand.